docs/modules/ROOT/pages/how-tos/configuring/adding-custom-tasks.adoc

```diff
@@ -0,0 +1,146 @@
+= Adding custom tasks
+
+If your pipeline uses xref:advanced-how-tos/using-trusted-artifacts.adoc[Trusted Artifacts],
+you can add custom tasks to the pipeline without compromising the chain of trust.
+These tasks can do just about anything you'd like: set dynamic arguments for the build, run unit tests, send emails, etc.
+With one major restriction: they *cannot modify your source repo or the prefetched content* before the build.
+Doing so would violate the link:https://enterprisecontract.dev/docs/ec-policies/release_policy.html#trusted_task__trusted[Trusted Tasks] Enterprise Contract rule.
+
+What this means in Trusted Artifacts terms:
+
+* To operate on your repo, your Task must declare a `SOURCE_ARTIFACT` parameter and use a special step to download this artifact
+  (refer to the ready-to-use examples below).
+* Your Task must *not* return a modified `SOURCE_ARTIFACT`.
+  It's ok if the task modifies the source repo as it executes, but the modifications will be discarded.
+  The build will use the original, unmodified `SOURCE_ARTIFACT`.
++
+NOTE: If you do return a modified `SOURCE_ARTIFACT` from a custom task and pass it to the build task,
+    Enterprise Contract will detect that the chain of trust contains an untrusted task and report the violation.
+
+.Prerequisites:
+
+. Your pipeline uses xref:advanced-how-tos/using-trusted-artifacts.adoc[Trusted Artifacts].
+
+.Procedure:
+
+. Implement the custom Task in the `.tekton/tasks/` directory in your repo.
++
+NOTE: The presence of the Task in `.tekton/**` is enough to make it available for use in your Pipelines without any extra configuration;
+    see link:https://pipelinesascode.com/docs/guide/resolver/[Pipelines-as-Code resolver].
```

Comment:
Another PR is proposing a different structure (for remote resolvers). Should we address the potential confusion?

https://github.com/konflux-ci/docs/pull/173/files#diff-7de75443c634564136fd1db89ef54619c2483dd2be6d8063cd0ffd764e96476dR73-R82

Reply:
Hmm, interesting. They are different use cases, but we probably do need to differentiate between them somewhere

That PR: defining a complete "remote" Pipeline (will be referenced from other repos)
This PR: adding a custom Task to a non-remote Pipeline

Reply:
Yeah. I don't feel like this PR needs to include both local and remote pipelines/tasks, but maybe just an acknowledgement the the other exists as well?

Should we just keep this PR scoped down to creating your own custom tasks? Then there can be a separate location where we talk about using remote tasks? These pages can be located next to each other.